### PR TITLE
19: Initialize front end app and create basic interface

### LINKED ISSUE #19: Initialize front-end app and create login page
**Description:**

- Initialize a Vue application with the required libraries to create a base interface. 
- Initialize dependency management and a way to start up the app and investigate the usefulness of containerization for the app.
- Create login page.

**Acceptance Criteria:**

- [x] Able to run `npm run serve` and app will open.
- [x] Login page is completed with username and password input fields, and login button.

### PR DESCRIPTION
**Pull request by ZackHolmberg**
# Summary of Changes:

- Initialized a Vue app for the frontend
- Added a mock login page
- Wrote a couple of reusable components 

# How to test
- You will vue CLI installed on your computer
- Navigate to the app/bisoncoin folder
- Run `npm install` to install dependencies
- Run `npm run serve` to run the app on a local port
- Open the address in the browser

Closes #19

### CODE REVIEW

File: src/app/bisoncoin/src/components/Button.vue [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+<style>
+.button {
+  background-color: #00a3e050;
+  width: 100px;
+  height: 50px;
+  border-radius: 25px;
+  box-shadow: 0 4px 8px 0 rgba(0, 0, 0, 0.2);
+  font-size: 16px;
+  border: none;
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/src/components/Button.vue]: Can remove this as it is being overwritten

File: src/app/bisoncoin/src/components/Button.vue [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+@Component
+export default class Button extends Vue {
+  @Prop() private label!: string;
+}
+</script>
+
+<style>
+.button {
+  background-color: #00a3e050;
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/src/components/Button.vue:18]: Should store all colours and text in separate asset files and then reference them in components and styles

File: src/app/bisoncoin/src/pages/HomePage.vue [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="bison">BisonCoin</div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+@Component
+export default class HomePage extends Vue {
+  @Prop() private msg!: string;
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/src/pages/HomePage.vue]: Can remove this, don't need it

File: src/app/bisoncoin/src/pages/HomePage.vue [hunk 4]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="home">
+    <h1>{{ msg }}</h1>
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/src/pages/HomePage.vue]: Remove use of the prop that we're removing

File: src/app/bisoncoin/src/pages/HomePage.vue [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+<script lang="ts">
+import { Component, Prop, Vue } from 'vue-property-decorator';
+
+@Component
+export default class HomePage extends Vue {
+  @Prop() private msg!: string;
+}
+</script>
+
+<!-- Add "scoped" attribute to limit CSS to this component only -->
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/src/pages/HomePage.vue]: Remove comment

File: src/app/bisoncoin/src/pages/LoginPage.vue [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    <img
+      width="300px"
+      height="300px"
+      alt="BisonCoin logo"
+      src="../assets/BisonCoin.png"
+    />
+    <TextInput label="Username" />
+    <TextInput label="Password" />
+    <Button label="Login" />
+    <a href="hello">Don't have an account? Create one now!</a>
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/src/pages/LoginPage.vue]: Updated HREF to say TODO instead. Additionally, move this hardcoded text into a separate file as mentioned prior

File: src/app/bisoncoin/tests/e2e/specs/test.js [hunk 7]
```diff
@@ -0,0 +1,8 @@
+// https://docs.cypress.io/api/introduction/api.html
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/tests/e2e/specs/test.js:1]: Should add a mock E2E test for future reference!

File: src/blockchain/blockchain.py [hunk 8]
```diff
@@ -0,0 +1,112 @@
+from hashlib import sha256
```
Comment (ZackHolmberg) [inline src/blockchain/blockchain.py]: Remove this file!

File: src/app/bisoncoin/package.json [hunk 9]
```diff
@@ -0,0 +1,39 @@
+{
```
Comment (ZackHolmberg) [inline src/app/bisoncoin/package.json:1]: Add this file to git ignore!

**Review by madisonfines — APPROVED** [review]

**Review by abhisachdev17 — CHANGES_REQUESTED** [review]
can you move the folder structure to be frontend > src > .. to separate frontend and backend folders.

**Comment by ZackHolmberg** [comment]
> can you move the folder structure to be frontend > src > .. to separate frontend and backend folders.

@abhisachdev17 The current folder structure is kind of trying to do this already, with src being src code parent folder, and then next 3 directories are `app` (frontend), `services`(middleware) and `blockchain` (could be called backend). Do you think this is satisfactory or did you have something different in mind?

**Comment by abhisachdev17** [comment]
> you

I was thinking of having frontend and backend in the root directory of the project, for separation right from the beginning.

**Review by abhisachdev17 — APPROVED** [review]

**Pull request merged**